### PR TITLE
docs: add tutorial on evaluating pretraining on custom detection data

### LINKED ISSUE #924: [DOC] Guide: evaluating pretraining/distillation on a custom detection dataset
### 📍 Location of the issue

- File/Section: New guide — no existing page covers this. Closest existing material: the FAQ entry ["Can I train on labeled images?"](https://docs.lightly.ai/train/stable/faq.html#can-i-train-on-labeled-images) and the tutorial [Object Detection with Ultralytics' YOLOv26](https://docs.lightly.ai/train/stable/tutorials/yolo/index.html)
- Link (if applicable): https://docs.lightly.ai/train/stable/faq.html#can-i-train-on-labeled-images

### ✏️ Suggested Improvement

When evaluating pretraining or distillation on a custom dataset, several setup choices can make results hard to interpret: the initialization of each training run, the labeled subset used for fine-tuning, random seeds, training budget, and whether validation and test images are excluded from the unlabeled pretraining set.

The FAQ documents the validation-leakage rule, and the YOLOv26 tutorial demonstrates a from-scratch vs. pretrained comparison. I could not find a single consolidated guide describing an endorsed, reproducible protocol for running this comparison on a user's own detection dataset.

**Proposal (documentation only):** an "Evaluating pretraining on your own data" guide, extending an existing object-detection workflow and documenting maintainer-recommended guidance for:

- How to choose and clearly report the compared configurations and their initialization
- How to match training budgets and which hyperparameters to report
- How to create reproducible, seeded labeled subsets
- When multiple seeds are warranted for small labeled subsets
- How to keep validation and test images out of the unlabeled pretraining set
- A results-table template covering raw metrics and run configuration — I can populate it with a reference run on VOC if that would be useful

The guide would not prescribe that pretraining must outperform a baseline; the goal is to make comparisons reproducible and interpretable.

**Non-goals:**

- Training orchestration, sweep tooling, or AutoML
- ROI, annotation-cost, or pass/fail claims
- Sample selection, near-duplicate detection, or dataset curation
- New dataset parsers, package APIs, or product features

A small example script for persisted, seeded splits could be a separate discussion later if there is interest.

### 💬 Additional Notes

Related discussions: #71, #371, #433, #471, #922. Each asks a variant of this question and was answered individually in its thread — from #371: "The best way of truly knowing whether pretraining helps is to actually do the whole fine-tuning."

**Questions:**

1. Does the work mentioned in #71 ("we are actively working on this") already cover this, or is there internal documentation I should align with?
2. Which reference workflow should the guide build on — `train_object_detection`, the Ultralytics YOLO tutorial, or another supported workflow?
3. For the comparison baselines, should the guide cover random initialization, public COCO-pretrained initialization, or both?
4. Would a docs-only PR be useful now, or is this outside the current roadmap?

I will wait for direction before opening a PR.

### PR DESCRIPTION
**Pull request by saud5150**
## What has changed and why?

Closes #924.

Adds a new docs page, **"Evaluating Pretraining on Your Own Detection Data"**
(`docs/source/tutorials/pretraining_evaluation/index.md`), following the structure
agreed in the issue thread. It consolidates the fair-comparison guidance that so far
only existed in individual issue answers (#71, #371, #433, #922) into one linkable
page.

The guide starts from the situation a reader is actually in — one directory of
unlabeled images, one of annotated ones — simulates that starting point from VOC as
explicitly labelled scaffolding, and holds the evaluation split out of the annotated
directory rather than borrowing a ready-made VOC split. The held-out set is placed
outside the pretraining directory, so leakage is structurally impossible rather than
a rule the reader has to remember. From there it covers the four configurations worth
comparing (including the head-initialization trap from #71 and pretraining on top of
COCO weights), what to hold fixed across runs, a runnable block with the pretrain and
fine-tune calls, a results-table template, and how to read the outcome without
overclaiming.

Two notes for review. The three rules under "Match the Training Budget and Run" —
identical held-out set per run, one checkpoint-selection rule, and not drawing
conclusions from a starved pretraining run — are my additions rather than anything
the team has stated; happy to drop or reword them. And the multi-seed guidance in the
final section is standard practice rather than LightlyTrain policy.

@liopeer — taking you up on your offer to train the configurations: the results table
ships as a template with the metric cells empty, so no numbers had to be invented from
short schedules. Happy to adjust the columns to whatever you run.

## How has it been tested?

- `uv run --frozen mdformat --check` clean on all touched files
- `uv run --frozen pytest docs/format_code.py::test_format_check_code_in_docs` —
  171 passed
- `cd docs && make docs` (sphinx with `--fail-on-warning`) builds clean; verified in
  the built HTML that the page renders, appears in the tutorials navigation, and the
  cross-links resolve
- Smoke-tested against a real VOC download, running the prerequisites script and
  dataset YAML extracted verbatim from the page: 4717 unlabeled / 800 train / 200
  held-out images with zero overlap between train and held-out, and
  `check_det_dataset` resolves the YAML to the intended directories with `nc=20`.
  The trainings themselves were not run.

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [x] Yes
- [ ] Not needed (internal change)

## Did you update the documentation?

- [x] Yes
- [ ] Not needed (internal change without effects for user)

### CODE REVIEW

File: docs/source/tutorials/pretraining_evaluation/index.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+## Choose What You're Comparing
+
+Decide — and write down — which configurations you are comparing *before* training
+anything. For YOLO-style detectors there are three meaningful starting points:
+
+| Configuration              | How to get it                                                                                   | What it tells you                                          |
+| -------------------------- | ----------------------------------------------------------------------------------------------- | ---------------------------------------------------------- |
+| Random initialization      | `model="ultralytics/yolo26s.yaml"`                                                              | Does pretraining beat no prior knowledge at all?           |
+| COCO-pretrained weights    | `model="yolo26s.pt"` directly in fine-tuning                                                    | Does pretraining beat the standard off-the-shelf start?    |
+| Pretraining on top of COCO | `model="ultralytics/yolo26s.pt"` in `lightly_train.pretrain`, then fine-tune the exported model | Does pretraining add value *on top of* the standard start? |
```
Comment (liopeer) [inline docs/source/tutorials/pretraining_evaluation/index.md]: The model strings are a bit confusing, since 2 of them can be used with LightlyTrain, but the 3rd can't.

File: docs/source/tutorials/pretraining_evaluation/index.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+```text
+datasets/VOC
+├── images
+│   ├── train2007      # ✅ include in pretraining
+│   ├── train2012      # ✅ include in pretraining
+│   ├── val2007        # ❌ never include
+│   ├── val2012        # ❌ never include
+│   └── test2007       # ❌ never include
+└── labels             # ignored by pretraining
+```
```
Comment (liopeer) [inline docs/source/tutorials/pretraining_evaluation/index.md]: I would start this a bit differently. Most people actually have:
1. one directory with non-annotated images
2. one directory with annotated images
We should show how they can build a correct split from that starting point, not from something like VOC that is already completely annotated. This also means that the 10% split you do below should rather be a prerequisite than part of the tutorial, otherwise the tutorials creates an artificial setting.

My suggestion:

Start with a "Prerequisites" section where you do your split on VOC (you remove some annotations, you leave a few annotated samples). State clearly that this is just so that we have a starting point with annotated+non-annotated samples.

File: docs/source/tutorials/pretraining_evaluation/index.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+- **Fine-tuning training images: yes.** Include them — pretraining works best when it
+  sees the full data distribution, and this will make your model better.
+- **Validation and test images: never.** Including them leads to data leakage: the
+  evaluation is no longer measuring generalization. Treat the unlabeled pretraining set
+  like a training split — any image in it must not be used for evaluation.
+- **Additional unlabeled images: yes.** The more the better, as long as they come from
+  the same domain.
+
+See the {ref}`FAQ <faq>` entries *"Can I train on labeled images?"* and *"How much data
```
Comment (liopeer) [inline docs/source/tutorials/pretraining_evaluation/index.md:169]: I think we should also backlink from the FAQ to this tutorial.

**Review by liopeer — CHANGES_REQUESTED** [review]
Definitely looking nice already! I would structure it however a bit differently to make the tutorial feel less *artificial*. Once it's restructure I can actually set up the benchmarks.

File: docs/source/tutorials/pretraining_evaluation/index.md [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+| Configuration              | Pretraining step (LightlyTrain)                               | Fine-tuning starts from (Ultralytics)              |
+| -------------------------- | ------------------------------------------------------------- | -------------------------------------------------- |
+| Random initialization      | —                                                             | `YOLO("yolo26s.yaml")`                             |
+| COCO-pretrained weights    | —                                                             | `YOLO("yolo26s.pt")`                               |
+| Pretraining on top of COCO | `pretrain(model="ultralytics/yolo26s.pt", data="my_dataset")` | `YOLO("out/.../exported_models/exported_last.pt")` |
+
+We recommend including the third configuration: pretraining does not have to replace
+COCO weights, it can build on them. Off-the-shelf initialization and pretraining on your
+own data are complementary, not mutually exclusive.
```
Comment (liopeer) [inline docs/source/tutorials/pretraining_evaluation/index.md]: This is now different from what was discussed. Now we're *only* pretraining on top of COCO weights. Also does not match with the warning that comes after.

File: docs/source/tutorials/pretraining_evaluation/index.md [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+  configurations is small, repeat the fine-tuning with 2–3 different seeds and report
+  the spread. A gap that is smaller than the run-to-run variation is not a conclusion.
+- **A null result is information, not failure.** If pretraining did not help, that is a
+  valid, reportable outcome for your dataset and budget — it does not mean the
+  comparison was wasted or that pretraining never helps.
+- **If you want to dig further**, the levers that most often change the outcome are:
+  training longer (see the
+  {ref}`recommended epochs for distillation <methods-distillation>`), pretraining on top
+  of the COCO weights if you started from scratch, using a larger model (small models
+  tend to profit less from pretraining), and revisiting the learning rate.
```
Comment (liopeer) [inline docs/source/tutorials/pretraining_evaluation/index.md]: Put into bullet points.

**Review by liopeer — CHANGES_REQUESTED** [review]

**Comment by saud5150** [comment]
Both fixed — the table's back to four rows so the head-init warning refers to
something again, and the levers are bulleted.

Going back over your restructure comment I'd only done half of it: the starting
point was two directories, but the eval set was still a ready-made VOC split, which
nobody has with their own data. The prerequisites now hold out 200 of the 1000
annotated images into `my_valset/`, outside `my_dataset/` so pretraining can't reach
it. I ran that section against a real VOC download before posting so it doesn't cost
you time later — 4717 unlabeled / 800 train / 200 held-out, zero overlap, and the
dataset YAML resolves to the right two directories. Didn't run the trainings
themselves; those are yours.

Two bugs fell out of that: `val` was pointing at `val2012`, which VOC.yaml lists
under `train` (ironic one to ship in a guide about split hygiene), and the script
assumed `./datasets` instead of reading `settings["datasets_dir"]`, so it would have
quietly copied nothing. I also added a short runnable block with the pretrain and
fine-tune calls, so the four configurations are directly executable rather than
reconstructed from the YOLO tutorial.

The three rules under "Match the Training Budget" are my additions rather than
anything you'd said — happy to drop any of them. Ready for another look, and for the
benchmark runs.

File: docs/source/tutorials/pretraining_evaluation/index.md [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+## Draw the Data Boundary
+
+Which images may enter the unlabeled pretraining pool?
+
+- **Your labeled training images: yes.** Pretraining works best when it sees the full
+  data distribution, and this will make your model better.
+- **Your held-out images: never.** Including them leads to data leakage: the evaluation
+  is no longer measuring generalization. Treat the pretraining pool like a training
+  split — any image in it must not be used for evaluation.
+- **Additional unlabeled images: yes.** The more the better, as long as they come from
```
Comment (liopeer) [inline docs/source/tutorials/pretraining_evaluation/index.md]: Suggestion: In general, the more the better, however also pretraining benefits from curating your dataset, i.e. making sure that dominating distributional modes get tamed and modes in the long tail are still represented after subsampling. I would check LightlyStudio for this.

Note: Don't use this word-by-word. But I would add something in that direction and actually link to the LightlyStudio GitHub.

File: docs/source/tutorials/pretraining_evaluation/index.md [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+| random_init             | random                     | —                         | 800 images / 50           | 0    | …        |
+| coco                    | COCO weights               | —                         | 800 images / 50           | 0    | …        |
+| pretrained_from_scratch | LightlyTrain (from random) | my_dataset / 100          | 800 images / 50           | 0    | …        |
+| pretrained_on_coco      | LightlyTrain (on COCO)     | my_dataset / 100          | 800 images / 50           | 0    | …        |
+
+For Ultralytics fine-tuning the metric comes from the `results.csv` of each run, as
+shown in the {ref}`YOLOv26 tutorial <tutorials-yolo>`. If you fine-tune with
+LightlyTrain's `train_object_detection` instead, the
+{ref}`benchmark command <quick-start-object-detection>` (in beta) evaluates a checkpoint
+on a validation split and writes a report with the run configuration included.
```
Comment (liopeer) [inline docs/source/tutorials/pretraining_evaluation/index.md]: Change to checking in the logs or in tensorboard. Additionally running the benchmark command would be wasteful.

**Review by liopeer — APPROVED** [review]
Very small final issues, otherwise good from my side! I will launch the actual comparison runs from my side, might take a few days until I get to gather and share the results.

Also it looks like you still need to sign the CLA before we can merge this.

File: docs/source/tutorials/pretraining_evaluation/index.md (hunk 6 above)
Comment (saud5150) [reply docs/source/tutorials/pretraining_evaluation/index.md]: Thanks for the curation tip — that one was a real gap!

**Comment by saud5150** [comment]
@liopeer — waiting on your comparison runs.
